lib/mirage_key.ml

```diff
@@ -91,16 +91,34 @@ type mode = [
   | `Qubes
 ]
 
+let first_mention = ref true
+
 let target_conv: mode Cmdliner.Arg.converter =
-  Cmdliner.Arg.enum [
-    "unix"  , `Unix;
-    "macosx", `MacOSX;
-    "xen"   , `Xen;
-    "virtio", `Virtio;
-    "hvt"    ,`Hvt;
-    "muen"  , `Muen;
-    "qubes" , `Qubes
-  ]
+  let parser, printer =
+    Cmdliner.Arg.enum [
+      "unix"  , `Unix;
+      "macosx", `MacOSX;
+      "xen"   , `Xen;
+      "virtio", `Virtio;
+      "hvt"   , `Hvt;
+      "muen"  , `Muen;
+      "qubes" , `Qubes;
+    ]
+  in
+  let filter_ukvm s =
+    let str =
+      if s = "ukvm" then begin
+        if !first_mention then begin
+          Logs.warn (fun m -> m "The ukvm target has been renamed to hvt. Please use `-t hvt` the next time.") ;
```

Comment:
How about:

    "The `ukvm' target has been renamed to `hvt', and support for `-t ukvm` will be removed in a future MirageOS release. Please change to using `-t hvt` at your earliest convenience."

...very polite and english, and also tells the user that we _will_ remove the backward compatibility at some point.

Reply:
Actually, that might be a bit misleading, maybe something like:

    "The `ukvm' target has been renamed to `hvt', and support for it will be removed in a future MirageOS release. Please reconfigure using `-t hvt' at your earliest convenience."

Also, will this do the right thing and print the warning on the initial "mirage clean" using the cached configuration?